backend/services/resource.py

```diff
@@ -144,22 +140,21 @@ def delete(self, user: User, id: int) -> None:
     def get_by_slug(self, user: User, search_string: str) -> list[Resource]:
         """
         Get a list of resources given a search string that the user has access to
-
         Parameters:
             user: a valid User model representing the currently logged in User
             search_string: a string to search resources by
-
         Returns:
             list[Resource]: list of resources relating to the string
-
         Raises:
             ResourceNotFoundException if no resource is found with the corresponding slug
         """
         query = select(ResourceEntity).where(
-            ResourceEntity.title.ilike(f"%{search_string}%"),
-            ResourceEntity.role == user.role,
-            ResourceEntity.group == user.group,
+            ResourceEntity.name.ilike(f"%{search_string}%"),
+            ResourceEntity.program.in_(user.program)
         )
         entities = self._session.scalars(query).all()
 
-        return [entity.to_model() for entity in entities]
+        if not entities:
+             raise ResourceNotFoundException(f"No resource found with matching slug: {search_string}")
```

Comment:
Instead of raising an error here, we should return an empty list because we don't want our API to return a 500 status code if the search results in nothing.

Reply:
So you can just remove the raising of this exception